### PR TITLE
Remove CatBoost integration for isolation.

### PR DESCRIPTION
**Pull request by y0z**
## Motivation & Description of the changes
- Progress https://github.com/optuna/optuna/issues/4484
  - Remove CatBoost integration from this repository for isolation
  - Use `optuna_integration.catboost` instead of `optuna.integration.catboost`

Please merge the following PR before merging this PR.
- https://github.com/optuna/optuna-integration/pull/61

### CODE REVIEW

**Comment by y0z** [comment]
@nabenabe0928 Could you review this PR and the related PR (https://github.com/optuna/optuna-integration/pull/61) ?

**Comment by HideakiImamura** [comment]
@nabenabe0928 Could you review this PR?

File: docs/source/reference/integration.rst [hunk 1]
```diff
@@ -144,8 +135,6 @@ We summarize the necessary dependencies for each integration.
 +-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------+
 | `Catalyst <https://github.com/optuna/optuna/blob/master/optuna/integration/catalyst.py>`_                                                                                         | catalyst                           |
 +-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------+
-| `CatBoost <https://github.com/optuna/optuna/blob/master/optuna/integration/catboost.py>`_                                                                                         | catboost                           |
-+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------+
```
Comment (nabenabe0928) [inline docs/source/reference/integration.rst:148]: Everything moved to optuna-integration is still here, e.g., shap, AllenNLP, so I think we can keep it.

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]
I left one comment, but otherwise the PR looks good to me.

File: docs/source/reference/integration.rst (hunk 1 above)
Comment (y0z) [reply docs/source/reference/integration.rst]: Fixed.

**Comment by y0z** [comment]
Thank you for your feedback. I applied your comment. PTAL.

**Review by nabenabe0928 — APPROVED** [review]
Thank you for the changes, LGTM!

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**